libs/providers/vmware.py

```
@@ -1837,16 +1857,63 @@ def get_vm_or_template_networks(
         Returns:
-            List of network mappings
+            List of network mappings in format [{"name": "network-name"}, ...]
         """
-        return inventory.vms_networks_mappings(vms=names)
+        try:
+            # Try Forklift inventory first (works for VMs)
+            return inventory.vms_networks_mappings(vms=names)
+        except ValueError as e:
+            # If VM not found in inventory, try direct vSphere query (for templates)
+            if "not found" in str(e):
+                LOGGER.info(f"VM(s) not found in Forklift inventory, querying vSphere directly for templates: {names}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)